Source/Orts.Parsers.OR/JsonReader.cs

```diff
@@ -194,9 +222,36 @@ public float AsTime(float defaultValue)
             }
         }
 
+        public Vector3 AsVector3(Vector3 defaultValue)
+        {
+            var vector3 = defaultValue;
+            switch (_reader.TokenType)
+            {
+                case JsonToken.StartArray:
+                    if (_reader.Read())
+                        vector3.X = AsFloat(0f);
+                    if (_reader.Read())
+                        vector3.Y = AsFloat(0f);
+                    if (_reader.Read())
+                        vector3.Z = AsFloat(0f);
+                    if (!_reader.Read() || _reader.TokenType != JsonToken.EndArray)
+                        goto default; // We did not have exactly 3 items in the array
+                    _path.Length = _pathPositions.Pop();
+                    return vector3;
+                default:
+                    TraceWarning($"Expected Vector3 (3 item array) in {Path}; got {_reader.TokenType}");
+
+                    // If the end of the array is not found in the right position, then parsing of subsequence objects also fails.
+                    TraceWarning($"Subsequent objects may be skipped");
```

Comment:
This warning should not be here

I can only think of two reasons subsequent objects may be skipped:

1.  A bug in `AsVector3` or another part of `JsonReason` leaves it in an incorrect state compared to the JSON
2. Mismatched object/array brackets in the JSON

For one, we should fix the bug, and for two there should already be another warning or error emitted due to the problem (it might be inscrutable but when you mismatch brackets that's what you get - both in SIMIS and JSON)

Reply:
Agreed